lib/modules/manager/npm/post-update/pnpm.ts

```diff
@@ -26,9 +27,18 @@ export async function generateLockFile(
       (upgrade) =>
         upgrade.depType === 'packageManager' && upgrade.depName === 'pnpm'
     );
+    let pnpmEngine;
```

Comment:
Config constraints should have highest prio. So only try to lookup when not already configured.